NPS_generation/functions.py

```diff
@@ -454,3 +456,15 @@ def NeutraliseCharges(mol, reactions=None):
             rms = AllChem.ReplaceSubstructs(mol, reactant, product)
             mol = rms[0]
     return mol
+
+
+def set_seed(seed):
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
```

Comment:
Important step for reproducibility that I missed in my first attempt (doesn't hurt to seed `torch.cuda.manual_seed` even if you don't have a gpu).